components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3543,7 +3561,10 @@ subroutine compute_rain_fall_velocity(qr_incld, rcldm, rhofacr, nr, nr_incld, mu
 
    call find_lookupTable_indices_3(dumii,dumjj,dum1,rdumii,rdumjj,inv_dum3,mu_r,lamr)
 
-   nr = nr_incld*rcldm
+   !nr is not used elsewhere in this function. Here, the updated value from generalized_sed
+   !is getting overwritten with the incld value which wasn't getting updated until this PR.
+   !And why is nr getting updated but not qr? Should delete nr from this function entirely.
+   !nr = nr_incld*rcldm 
```

Comment:
@PeterCaldwell nr is updated here because get_rain_dsd2 updates its nr input ("lambda limiters for rain"), which is nr_incld from the caller's perspective. So then this line propagates that modification to nr. The same applies to nc in cloud_sedimentation. So these lines should not be commented out.

Reply:
@ambrad 
That is correct: s/r `get_rain_dsd2` updates the `*_incld` values for the calculation of `V_*`.
In the `calc_first_order_upwind_step` s/b, the flux is calculated using the cell-avg values, thus needed to be updated together in `get_rain_dsd2`
The same goes for the `get_cloud_dsd2`.
Man, that pointer coding is efficient but confusing.

Reply:
Ok, it makes sense that since nr_incld was used to update lamr and mu_r so we need to enforce consistency between nr_incld and nr. I think the nr_incld=nr/rcldm calculation at the end of the CFL substepping loop now accomplishes that, though. I'd like a second opinion on whether I'm right...

Reply:
@PeterCaldwell 
Yes, but please note that `nr_incld` is not only used to update other vars, but also it's being update by the lambda limiters (as Andrew mention).

For me the focal point is in calculating the fluxes (in `*_upwind_step`):
`fluxes(i)%p(k) = vs(i)%p(k) * qnx(i)%p(k) * rho(k)`
Here `qnx(i)` is a cell-avg value, which hasn't been fully updated by the most recent cell-avg value.

Reply:
Ah, thanks @JS-WRF-SBM . I didn't fully absorb Andrew's comment. Yes, I agree these lines are needed. I'll make the change and add a comment.